compiler/next/lib/frontend/Parser/chapel.ypp

```diff
@@ -1315,17 +1315,35 @@ extern_block_stmt:
 
 loop_stmt:
   TDO stmt TWHILE expr TSEMI
```

Comment:
Here `stmt` could be a block or it might not be.
I think your PR needs adjustment to better handle and test the case when `stmt` is not a block.
E.g. this compiles and runs with the production compiler

``` chapel
var i = 0;

proc foo() {
  writeln(i);
  i += 1;
}

do
  foo();
while i < 3;
```

I think this no-curly-braces form of do/while is of questionable value but it exists so we should handle it correctly. (If you think it should not exist then make a separate issue to discuss that).

One result of this is that the do/while AST node *should* have a `usesDo` field.

Reply:
I don't think it should be called `usesDo` (to me this doesn't make sense - a DoWhile always uses do). I think it should have a method called something `isBodyBlock()`.

Reply:
I already have a test for the no-curly-braces form, though?

Reply:
yes